task_sdk/src/airflow/sdk/execution_time/comms.py

```diff
@@ -138,7 +138,9 @@ class PutVariable(BaseModel):
     type: Literal["PutVariable"] = "PutVariable"
 
 
+RTIFPayload = RootModel[dict[str, Optional[str]]]
```

Comment:
So in this case we can't use a root model, as we need a `type` parameter on the request to know what kind of object we get. (I think!)

Reply:
Right, the discriminator wouldn't work then. 
Let me try and make this a BaseModel

Reply:
Or I am wondering if we can consume the _generated.py stuff for `RTIFPayload` and extend that. Probably pydantic will complain

Reply:
I think in this case it needs to be a BaseModel so that it can have a `type` field in-order to get the the discriminator to work

```
FAILED task_sdk/tests/execution_time/test_supervisor.py::TestWatchedSubprocess::test_reading_from_pipes - TypeError: 'dict' is not a valid discriminated union variant; should be a `BaseModel` or `dataclass`
```

Reply:
Yeah, converting to a BaseModel. Should be ok on the server side with RootModel